SYCL/ESIMD/spec_const/Inputs/spec_const_common.hpp

```diff
@@ -0,0 +1,89 @@
+//==--------------- spec_const_common.h  - DPC++ ESIMD on-device test -----===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// The test checks that ESIMD kernels support specialization constants for all
+// basic types, particularly a specialization constant can be redifined and
+// correct new value is used after redefinition.
+
+#include "esimd_test_utils.hpp"
+
+#include <CL/sycl.hpp>
+#include <CL/sycl/INTEL/esimd.hpp>
+
+#include <iostream>
+#include <vector>
+
+using namespace cl::sycl;
+
+template <typename AccessorTy>
+ESIMD_INLINE void do_store(AccessorTy acc, int i, spec_const_t val) {
+  using namespace sycl::INTEL::gpu;
+#if STORE == 0
+  // bool
+  scalar_store(acc, i, val ? 1 : 0);
+#elif STORE == 1
```

Comment:
If I'm not missing anything, this splitting into 3 cases does not seem to bring any additional value to the test. Suggest to use scalar_store unconditionally.

Reply:
scatter function, that is used in scalar_store, can only be used with types whose size is no larger than 4:
> llvm/build/install/bin/sycl/CL/sycl/INTEL/esimd/esimd_memory.hpp +338
` typename sycl::detail::enable_if_t<(sizeof(T) <= 4) &&`